cmd/clusterctl/pkg/client/cluster/inventory.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/config"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/internal/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// InventoryClient has methods to work with provider inventory in the cluster.
+type InventoryClient interface {
+	// EnsureCRD installs the CRD required for creating inventory items, if necessary.
+	EnsureCRD() error
```

Comment:
Does this install only one CRD?

Reply:
As of today, it installs only one CRD that is generated from the `cmd/clusterctl/api` folder, but the code is ready to install all the ùcrd that will be eventually added to such folder (e.g. TBD if one is needed for the version matrix,I'm prototyping now on updates 😉).

NB. In order to provide a simpler out-of-the-box experience, the CRDs in the `cmd/clusterctl/api` are embedded in the clusterctl binary.

Reply:
What about renaming this to something like `EnsureCustomResourceDefinitions`? I like being explicit :)

Reply:
done